### PR TITLE
Convert grading_job_status sproc to model function

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Also adds some types to socket messages passed between by question sockets.

### CODE REVIEW

File: apps/prairielearn/src/lib/externalGradingSocket.ts [hunk 1]
```diff
@@ -88,18 +96,7 @@ export function connection(socket: Socket) {
...
-          answerPanel: panels.answerPanel,
-          submissionPanel: panels.submissionPanel,
-          extraHeadersHtml: panels.extraHeadersHtml,
-          questionScorePanel: panels.questionScorePanel,
-          assessmentScorePanel: panels.assessmentScorePanel,
-          questionPanelFooter: panels.questionPanelFooter,
-          questionNavNextButton: panels.questionNavNextButton,
-        });
-      },
+      (panels) => callback(panels),
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/externalGradingSocket.ts:99]: Turns out this is also fixing a bug: the `scorePanel` was missing in this update. I tested in master, and indeed the score panels are not updated when the result of the external grader is retrieved, while this PR fixes that.

File: apps/prairielearn/src/lib/question-render.js [hunk 2]
```diff
@@ -42,6 +41,9 @@ import { writeCourseIssues } from './issues.js';
 import * as manualGrading from './manualGrading.js';
 import { getQuestionCourse, ensureVariant } from './question-variant.js';
 
+/** @import { SubmissionPanels } from './question-render.types.js'  */
+/** @import { SubmissionForRender } from '../components/SubmissionPanel.html.js' */
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/question-render.js:45]: This is new in TS 5.5, and I think it's pretty neat...

File: apps/prairielearn/src/lib/question-render.js [hunk 3]
```diff
@@ -602,7 +589,6 @@ export async function renderPanelsForSubmission({
   /** @type {SubmissionPanels} */
   const panels = {
     submissionPanel: null,
-    scorePanel: null,
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/question-render.js:605]: This value was never used anywhere. The score panels are actually `questionScorePanel` and `assessmentScorePanel`.

**Review by nwalters512 — APPROVED** [review]
Tested this locally, LGTM.

**Pull request merged**